src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -672,6 +709,49 @@ int XrdHttpProtocol::Process(XrdLink *lp) // We ignore the argument here
 
   }
 
+
+#if OPENSSL_VERSION_NUMBER >= 0x10100010L
+  if (postheaderwait) {
+    postheaderwait = false;
+    if (SSL_verify_client_post_handshake(ssl) != 1) {
```

Comment:
We tried very very hard to keep all OpenSSL calls from not spreading outside of the XrdTls class. This destroys that principal. Could you please come up with a containment mechanism that moves this code as a function of TlsContext or TlsSocket, as appropriate? This really eases the maintenance burden,.

Reply:
This is all manipulation of the SSL socket so the natural place for this would be to live in `XrdTlsSocket`.

However, there's zero use of `XrdTlsSocket` in `XrdHttp` - it utilizes the SSL context directly.  Introducing and using it for only this small part of the handshake doesn't seem practical.  If we bring `XrdTlsSocket` in, it wants to own the SSL context object and manipulates its internals.

So, short of a reimplementation of the guts of the XrdHttpProtocol object to use XrdTlsSocket, is there any way to provide a containment mechanism?

Reply:
I tend to agree with @bbockelm here, we need to live with the SSL calls as they are.